core/app/models/spree/gift_card_batch.rb

```diff
@@ -0,0 +1,67 @@
+module Spree
+  class GiftCardBatch < Spree.base_class
+    extend DisplayMoney
+    include Spree::SingleStoreResource
+
+    LIMIT = ENV.fetch('GIFT_CARD_BATCH_LIMIT', 50_000).to_i
```

Comment:
```suggestion
    LIMIT = ENV.fetch('GIFT_CARD_BATCH_LIMIT', 50_000).to_i
```

@brozek95 let's move this to configuration - https://github.com/spree/spree/blob/main/core/lib/spree/core/configuration.rb , let's nit use ENVs for that